src/crypto/CHIPCryptoPALmbedTLS.cpp

```diff
@@ -648,6 +722,26 @@ CHIP_ERROR P256PublicKey::ECDSA_validate_hash_signature(const uint8_t * hash, co
 CHIP_ERROR P256Keypair::ECDH_derive_secret(const P256PublicKey & remote_public_key, P256ECDHDerivedSecret & out_secret) const
 {
 #if defined(MBEDTLS_ECDH_C)
+
+#if defined(MBEDTLS_USE_TINYCRYPT)
+    CHIP_ERROR error     = CHIP_NO_ERROR;
+    int result           = 0;
+    size_t secret_length = (out_secret.Length() == 0) ? out_secret.Capacity() : out_secret.Length();
+
+    const mbedtls_uecc_keypair * keypair = to_const_keypair(&mKeypair);
+
+    VerifyOrExit(mInitialized, error = CHIP_ERROR_INCORRECT_STATE);
+
+    result = uECC_shared_secret(remote_public_key.ConstBytes() + 1, keypair->private_key, Uint8::to_uchar(out_secret));
```

Comment:
The +1 needs to be explained

Reply:
Sure, I'll write a comment. The first byte is the compression type, the X and Y coordinates follow.

Reply:
Yes, it has to be expressed that tinycrypt expects fully padded  raw uncompressed points everywhere, but in this API we always work with SEC1 uncompressed points that have this 04 prefix